### DISCUSSION TITLE
Object detection application giving malloc() error while running

### DISCUSSION THREAD

**Discussion by AmarRobo** [AxisCommunications/acap-native-sdk-examples#113] [General] [answered]
Object Detection app needs several start and stops before it gets running successfully. It usually takes 8 to 10 times of restarting the app before the malloc() error resolves. I have additional SD card on the camera. Is there any particular reason for this behavior?
I am hoping to have an object detection app that runs on start up and this behavior will most probably make it impossible

**Comment by Corallo** [comment, marked as answer]
Hello @amararjun 

Can you describe more this `malloc` error that you encounter?

**Reply by AmarRobo** [reply]
The log of the application shows something like this
![malloc() error](https://user-images.githubusercontent.com/27479245/221128635-8cc7ed12-93ae-41dc-95bd-913579d9e39a.png)
I think the app stops running  after this error.
And, I try to start and stop the application few times until the app runs without the error.

**Reply by Corallo** [reply]
You have probably done some changes that make that malloc to fail, I don't think it is an issue of the example.
The malloc is used to allocate space to read the label file.  Is it possible you are using a wrong label file, or you provided the wrong parameter in the manifest?

Also, I am going to move this discussion in the native repository as well

**Reply by AmarRobo** [reply]
@Corallo Thank you. I did not explain the situation properly. I am trying MobileNet SSD trained on a custom dataset for object detection. Yes, I have a new label file which has only seven entries ( seven classes). In this situation, can you suggest how to avoid the error? Why is it that the error   disappear after few tries ?

**Reply by Corallo** [reply]
Memory errors are not always consistently reproducible, it could be that it works once in a while when the memory of the device is in a specific state.

Have you updated the [Line 13 of the manifest](https://github.com/AxisCommunications/acap-native-sdk-examples/blob/main/object-detection/app/manifest.json.cpu#L13) replacing 80 with the number of your classes?
Note, I linked the cpu version of the manifest, if you are using the accelerator, you should it in also the other manifest file

**Reply by AmarRobo** [reply]
Thank you for the information. 
The 80 in  line 13 is supposed to be the image quality, is it not? [argparse.c](https://github.com/AxisCommunications/acap-native-sdk-examples/blob/80a62d431fbc1a0770dd7f914eab177618d2f3f8/object-detection/app/argparse.c#L104)

**Reply by Corallo** [reply]
Right, my bad, I got confused.

The number of label is actually established by the size of the label file.
So your crash is a bit strange.

Have you made other changes besides the model and the content of the labels file?

**Reply by AmarRobo** [reply]
@Corallo Thank you for the cue. A newline was required at the end of the labels.txt file. The malloc() error got solved.
